### PR TITLE
Add Git and OCI artifact inspection with enhanced validation

### LINKED ISSUE #61: inspect: display OCI manifest digest and Git commit SHA
## Summary

`striatum inspect` shows artifact metadata (name, version, kind, entrypoint, files, dependencies) but does not display the OCI manifest digest or Git commit SHA of the remote artifact.

## Current behavior

```
$ striatum inspect localhost:5050/skills/my-skill:1.0.0
Name:         my-skill
Version:      1.0.0
Kind:         Skill
Entrypoint:   SKILL.md
Files:        SKILL.md
```

## Expected behavior

For OCI artifacts:
```
$ striatum inspect localhost:5050/skills/my-skill:1.0.0
Name:         my-skill
Version:      1.0.0
Kind:         Skill
Digest:       sha256:e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855
Entrypoint:   SKILL.md
Files:        SKILL.md
```

For Git artifacts:
```
$ striatum inspect git:https://github.com/example/skill.git@main
Name:         my-skill
Version:      1.0.0
Kind:         Skill
Commit:       a1b2c3d4e5f6a1b2c3d4e5f6a1b2c3d4e5f6a1b2
Entrypoint:   SKILL.md
Files:        SKILL.md
```

## Context

- `ResolveDigest` already exists in `pkg/oci/inspect.go` and can resolve the OCI manifest digest via `target.Resolve()`.
- `gitbackend.Backend.ResolveCommit()` already resolves Git commit SHAs.
- The `inspect` CLI command (`internal/cli/inspect.go`) already calls `resolveTargetAndRef` which gives access to the target — it just needs additional calls to display the digest/commit.

This would allow callers to capture digests/commits for pinning in `artifact.json` dependency declarations (the `digest` and `commit` fields added in v0.2.0).

### PR DESCRIPTION
**Pull request by hbelmiro**
Resolves #61

### Description

This pull request introduces support for inspecting Git and OCI artifacts with improved validation and handling of references. Key changes include:

- **Git Artifact Support**: Added `git:<url>@<ref>` support for the `inspect` and `install` commands, handling explicit commit resolutions and unique staging directories.
- **OCI Digest Parsing**: Enhanced parsing of OCI references with optional digest handling, including validation and support for immutable digests.
- **Validation Improvements**: Introduced safety checks for artifact names and versions, preventing path traversal and unsafe metadata inclusion.
- **Refactoring and Tests**: Refactored inspection logic for better modularity and added comprehensive unit tests to cover edge cases, including ambiguous Git references, validation failures, and digest resolution.

### Checklist

- [ ] Include unit tests to validate the changes made.
- [ ] Update documentation to reflect the updates in artifact handling and validation.
- [ ] Refactor code for better modularity and maintainability.
- [ ] Ensure compatibility with existing tools and workflows.

### Additional Notes

- Updates were made to the `pull`, `inspect`, and `install` commands to support digest and commit resolution with improved error handling.
- Dependencies were updated with version bumps to address security and compatibility concerns.

### CODE REVIEW

File: internal/cli/inspect.go [hunk 1]
```diff
@@ -2,44 +2,92 @@ package cli
...
+	}
+	m, err := oci.Inspect(cmd.Context(), target, ref)
+	if err != nil {
+		return fmt.Errorf("inspect artifact manifest and metadata: %w", err)
+	}
+	digest, err := oci.ResolveDigest(cmd.Context(), target, ref)
+	if err != nil {
+		return fmt.Errorf("resolve digest: %w", err)
+	}
+	printManifest(cmd.OutOrStdout(), m, digest, "")
```
[the remark this answers is not in the document]
Comment (hbelmiro) [reply internal/cli/inspect.go:55]: Fixed. `inspectOCI` now uses the digest returned by `resolveTargetAndRef` when non-empty and only falls back to `ResolveDigest` when no pinned digest was provided.

File: internal/cli/inspect.go [hunk 2]
```diff
@@ -2,44 +2,92 @@ package cli
...
+	backend := &gitbackend.Backend{}
+	m, err := backend.Inspect(cmd.Context(), gitDep)
+	if err != nil {
+		return fmt.Errorf("inspect git artifact: %w", err)
+	}
+	commit, err := backend.ResolveCommit(cmd.Context(), gitDep)
+	if err != nil {
+		return fmt.Errorf("resolve git commit: %w", err)
+	}
+	printManifest(cmd.OutOrStdout(), m, "", commit)
```
[the remark this answers is not in the document]
Comment (hbelmiro) [reply internal/cli/inspect.go:78]: Fixed. `inspectGit` now resolves the commit first, pins it on `gitDep.Commit`, and then inspects by that commit so the output is always consistent.

File: internal/cli/inspect_test.go [hunk 3]
```diff
@@ -109,6 +111,197 @@ func TestInspect_InvalidRef_Errors(t *testing.T) {
...
+			"GIT_CONFIG_NOSYSTEM=1", "GIT_CONFIG_GLOBAL="+os.DevNull,
+		)
+		out, err := cmd.CombinedOutput()
+		if err != nil {
+			t.Fatalf("%v failed: %s\n%s", args, err, out)
+		}
+	}
+
+	run(dir, "git", "init", "--bare", "-b", "master", bareDir)
+	run(dir, "git", "clone", bareDir, workDir)
```
[the remark this answers is not in the document]
Comment (hbelmiro) [reply internal/cli/inspect_test.go]: Fixed. Extracted the shared helper to `git_test_helpers_test.go` and made `setupGitRepo` in `install_test.go` delegate to it.

File: internal/cli/inspect.go [hunk 4]
```diff
@@ -2,44 +2,95 @@ package cli
...
+	if err != nil {
+		return fmt.Errorf("resolve reference: %w", err)
+	}
+	if digest == "" {
+		digest, err = oci.ResolveDigest(cmd.Context(), target, ref)
+		if err != nil {
+			return fmt.Errorf("resolve digest: %w", err)
+		}
+	}
+	m, err := oci.Inspect(cmd.Context(), target, ref)
```
[the remark this answers is not in the document]
Comment (hbelmiro) [reply internal/cli/inspect.go]: Fixed. When a digest is provided, `inspectOCI` now inspects by digest instead of by tag, ensuring the manifest matches the pinned digest.

File: internal/cli/inspect.go [hunk 5]
```diff
@@ -2,44 +2,99 @@ package cli
...
+	if digest != "" {
+		inspectRef = digest
+	}
+	m, err := oci.Inspect(cmd.Context(), target, inspectRef)
+	if err != nil {
+		return fmt.Errorf("inspect artifact manifest and metadata: %w", err)
+	}
+	printManifest(cmd.OutOrStdout(), m, digest, "")
+	return nil
+}
```
[the remark this answers is not in the document]
Comment (hbelmiro) [reply internal/cli/inspect.go:57]: Fixed. `inspectOCI` now validates the digest format with `artifact.IsValidDigest` before using it, returning a clear error for malformed digests.

File: internal/cli/inspect.go [hunk 6]
```diff
@@ -2,44 +2,99 @@ package cli
...
-			_, _ = fmt.Fprintf(cmd.OutOrStdout(), "Files:        %s\n", strings.Join(m.Spec.Files, ", "))
-			if len(m.Dependencies) > 0 {
-				_, _ = fmt.Fprintln(cmd.OutOrStdout(), "Dependencies:")
-				for _, d := range m.Dependencies {
-					_, _ = fmt.Fprintf(cmd.OutOrStdout(), "  - %s [%s]\n", d.CanonicalRef(), d.Source())
-				}
+
+			if strings.HasPrefix(reference, "git:") {
+				return inspectGit(cmd, reference)
 			}
```
[the remark this answers is not in the document]
Comment (hbelmiro) [reply internal/cli/inspect.go]: Fixed. `args[0]` is now trimmed with `strings.TrimSpace` before routing.

File: internal/cli/inspect.go [hunk 7]
```diff
@@ -2,44 +2,99 @@ package cli
...
+	}
+	inspectRef := ref
+	if digest != "" {
+		inspectRef = digest
+	}
+	m, err := oci.Inspect(cmd.Context(), target, inspectRef)
+	if err != nil {
+		return fmt.Errorf("inspect artifact manifest and metadata: %w", err)
+	}
+	printManifest(cmd.OutOrStdout(), m, digest, "")
```
[the remark this answers is not in the document]
Comment (hbelmiro) [reply internal/cli/inspect.go:55]: Acknowledged. This is a performance optimization (saves one round-trip for remote registries). The current approach is correct and the extra resolve is cheap relative to the full manifest fetch. Deferring to a follow-up if profiling shows it matters.

File: internal/cli/inspect_test.go [hunk 8]
```diff
@@ -109,6 +110,140 @@ func TestInspect_InvalidRef_Errors(t *testing.T) {
...
+			entryIdx = i
+		}
+	}
+	if digestIdx == 0 {
+		t.Fatalf("Digest line not found in output:\n%s", got)
+	}
+	if kindIdx >= digestIdx || digestIdx >= entryIdx {
+		t.Errorf("Digest should appear between Kind and Entrypoint; Kind=%d Digest=%d Entrypoint=%d\n%s",
+			kindIdx, digestIdx, entryIdx, got)
+	}
```
[the remark this answers is not in the document]
Comment (hbelmiro) [reply internal/cli/inspect_test.go:168]: Fixed. Indices are now initialized to `-1` and explicitly checked for all expected lines before the ordering assertion.

File: internal/cli/inspect_test.go [hunk 9]
```diff
@@ -109,6 +110,140 @@ func TestInspect_InvalidRef_Errors(t *testing.T) {
...
+			entryIdx = i
+		}
+	}
+	if commitIdx == 0 {
+		t.Fatalf("Commit line not found in output:\n%s", got)
+	}
+	if kindIdx >= commitIdx || commitIdx >= entryIdx {
+		t.Errorf("Commit should appear between Kind and Entrypoint; Kind=%d Commit=%d Entrypoint=%d\n%s",
+			kindIdx, commitIdx, entryIdx, got)
+	}
```
[the remark this answers is not in the document]
Comment (hbelmiro) [reply internal/cli/inspect_test.go:212]: Fixed. Same as the digest test: indices initialized to `-1` with explicit presence check.

File: internal/cli/git_test_helpers_test.go [hunk 10]
```diff
@@ -0,0 +1,68 @@
...
+			t.Fatal(err)
+		}
+	}
+
+	manifest := `{
+  "apiVersion": "striatum.dev/v1alpha2",
+  "kind": "Skill",
+  "metadata": {"name": "git-skill", "version": "1.0.0"},
+  "spec": {"entrypoint": "SKILL.md", "files": ["SKILL.md"]}
+}`
```
[the remark this answers is not in the document]
Comment (hbelmiro) [reply internal/cli/git_test_helpers_test.go]: Fixed. `setupLocalGitRepo` now derives the manifest version from `tagName` by trimming a leading `v` prefix, so `v2.0.0` produces `version: "2.0.0"`.
